### PR TITLE
fix(docker): support nvcr.io tag listing

### LINKED ISSUE #17304: No docker auth found - returning
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### Please select which platform you are using if self-hosting.

_No response_

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

After upgrading to 32.163.0 in the app, we've seen thousands of `No docker auth found - returning` warnings.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

### PR DESCRIPTION
**Pull request by steved**
## Changes

Support nvcr.io image registry tag listing.

## Context

Updating an nvcr.io image (`nvidia/k8s-device-plugin` in my particular case) fails with the following error:
> DEBUG: Failed to get authHeaders for getTags lookup
> DEBUG: Failed to look up dependency nvcr.io/nvidia/k8s-device-plugin (nvcr.io/nvidia/k8s-device-plugin)(packageFile="[...]/Dockerfile", dependency="nvcr.io/nvidia/k8s-device-plugin")

nvcr.io seems to deviate slightly from other image registries in that it does not return a `service` with the `Www-authenticate` header:

```
$ curl -v https://nvcr.io/v2/ |& grep Www
< Www-Authenticate: Bearer realm="https://nvcr.io/proxy_auth",scope=""
```

Returning an empty service query param back works fine:

```
$ curl -I 'https://nvcr.io/proxy_auth?service=&scope=repository:nvcr.io/nvidia/k8s-device-plugin:pull'
HTTP/1.1 200
...
```

The original change was made in https://github.com/renovatebot/renovate/pull/11942 and does not explain why `service` must be a string.

- closes #17304

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/docker/index.ts [hunk 1]
```diff
@@ -159,7 +158,11 @@ export async function getAuthHeaders(
       scope = authenticateHeader.params.scope;
     }
 
-    const authUrl = `${authenticateHeader.params.realm}?service=${authenticateHeader.params.service}&scope=${scope}`;
+    let service = authenticateHeader.params.service;
+    if (!is.string(service)) {
+      service = '';
+    }
```
Comment (JamieMagee) [inline lib/modules/datasource/docker/index.ts:164]: nit:

```suggestion
    const service = authenticateHeader.params.service ?? '';
```

But I think it's better to have it as a constant. What do you think?

**Comment by JamieMagee** [comment]
Interesting! That change was originally made to satisfy the types required later. All the documentation I had found pointed to it being a required parameter:

Docker[^1]:
```
$ curl -s -I https://index.docker.io/v2/ | grep -Fi www-authenticate
www-authenticate: Bearer realm="https://auth.docker.io/token",service="registry.docker.io"
```
GCR:
```
$ curl -s -I https://mirror.gcr.io/v2/ | grep -Fi www-authenticate
www-authenticate: Bearer realm="https://mirror.gcr.io/v2/token",service="mirror.gcr.io"
```
MCR[^2]

[^1]: https://docs.docker.com/registry/spec/auth/token/
[^2]: https://docs.microsoft.com/en-us/rest/api/containerregistry/access-tokens/get-from-login?tabs=HTTP

**Review by JamieMagee — APPROVED** [review]

**Review by viceice — CHANGES_REQUESTED** [review]
see open conversation

File: lib/modules/datasource/docker/index.ts (hunk 1 above)
Comment (steved) [reply lib/modules/datasource/docker/index.ts:164]: I had it as that originally, but the [linter](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/restrict-template-expressions.md) was complaining that `params.service` could be a `string` or `string[]`:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/auth-header/index.d.ts#L9

Happy to change it, whichever is best.

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**